libs/tools/src/generators/replace-ui-version/generator.ts

```diff
@@ -0,0 +1,76 @@
+import { type Tree, formatFiles, readJsonFile, updateJson } from '@nx/devkit';
+import { readdir, stat } from 'node:fs/promises';
+import { join } from 'node:path';
+import process from 'node:process';
+
+async function recursivelyFindRelativePackageJsonFilePaths(startingDir: string): Promise<string[]> {
+	let results = [];
+	const list = await readdir(startingDir);
+	for (const file of list) {
+		const filePath = join(startingDir, file);
+		const fileStat = await stat(filePath);
+		if (fileStat.isDirectory()) {
+			results = results.concat(await recursivelyFindRelativePackageJsonFilePaths(filePath));
+		} else if (file === 'package.json') {
+			results.push(filePath);
+		}
+	}
+	return results;
+}
+
+const getSpartanDependencyKeys = (dependencies?: Record<string, string>): string[] =>
+	Object.keys(dependencies ?? {}).filter((key) => key.startsWith('@spartan-ng'));
+
+const replaceUiVersionInCliVersionsFile = (tree: Tree, oldVersion: string, newVersion: string) => {
+	const filePath = `libs/cli/src/generators/base/versions.ts`;
+	let contents = tree.read(filePath).toString();
+	contents = contents.replaceAll(oldVersion, newVersion);
```

Comment:
It is not the case now, but does this have the potential to replace versions of other packages too in the future if some other const has the same version as spartan oldVersion?

Reply:
Good point. We might want to make this a little smarter and only change versions with spartan in their variable name

Reply:
@elite-benni added some regex to handle this better and covered it by a unit test to make sure we don't mess up other versions as you pointed out 👍